Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -768,34 +768,66 @@ extension Tensor where Scalar: TensorFlowFloatingPoint {
+    ///  * `reflect` - mirror values along padding dimensions, excluding the edge value.
+    ///  * `symmetric` - mirror values along padding dimensions, including the edge value.
+    enum PaddingMode {
+        case constant(Scalar)
+        case reflect
+        case symmetric
+    }
+
+    /// Returns a tensor padded with constant according to the specified padding sizes.
     @inlinable
-    @differentiable(wrt: self, vjp: _vjpPadded(forSizes:with:) where Scalar: TensorFlowFloatingPoint)
+    @differentiable(wrt: self where Scalar: TensorFlowFloatingPoint)
     func padded(forSizes sizes: [(before: Int, after: Int)], with value: Scalar = 0) -> Tensor {
+        return self.padded(forSizes: sizes, with: .constant(value))
+    }
+
+    /// Returns a padded tensor according to the specified padding sizes and mode.
+    @inlinable
+    @differentiable(wrt: self, vjp: _vjpPaddedWithMode(forSizes:with:) where Scalar: TensorFlowFloatingPoint)
+    func padded(forSizes sizes: [(before: Int, after: Int)], with mode: PaddingMode) -> Tensor {
         let paddings = Tensor<Int32>(
             shape: [sizes.count, 2],
             scalars: sizes.flatMap { [Int32($0.before), Int32($0.after)] })
-        return Raw.padV2(self, paddings: paddings, constantValues: Tensor(value))
+        switch mode {
+        case .constant(let constantValue):
+            return Raw.padV2(self, paddings: paddings, constantValues: Tensor(constantValue))
+        case .reflect:
+            return Raw.mirrorPad(self, paddings: paddings, mode: Raw.Mode5.reflect)
+        case .symmetric:
+            return Raw.mirrorPad(self, paddings: paddings, mode: Raw.Mode5.symmetric)
+        }
     }
 }
 
 internal extension Tensor where Scalar: TensorFlowFloatingPoint {
     @inlinable
-    func _vjpPadded(
+    func _vjpPaddedWithMode(
         forSizes sizes: [(before: Int, after: Int)],
-        with value: Scalar
+        with mode: PaddingMode
     ) -> (Tensor, (Tensor) -> Tensor) {
-        let result = padded(forSizes: sizes, with: value)
+        let result = padded(forSizes: sizes, with: mode)
         return (result, { [rank = rankTensor, shape = shapeTensor] v in
             let paddings = Tensor<Int32>(
                 shape: [sizes.count, 2],
                 scalars: sizes.flatMap { [Int32($0.before), Int32($0.after)] })
-            let padBefore = Raw.slice(paddings,
-                begin: Tensor<Int32>([0, 0]),
-                size: Tensor<Int32>(stacking: [rank, Tensor<Int32>(1)]))
-            let begin = padBefore.reshaped(to: [-1])
-            return Raw.slice(v, begin: begin, size: shape)
+            switch mode {
+            case .constant(_):
+                let padBefore = Raw.slice(paddings,
+                        begin: Tensor<Int32>([0, 0]),
+                        size: Tensor<Int32>(stacking: [rank, Tensor<Int32>(1)]))
+                let begin = padBefore.reshaped(to: [-1])
+                return Raw.slice(v, begin: begin, size: shape)
```

Comment:
You can use [`slice(lowerBounds:sizes:)`](https://github.com/tensorflow/swift-apis/blob/f0a6da60bcd8192b958d0b008e9df2a7d0e57547/Sources/TensorFlow/Operators/Basic.swift#L828). We recommend using Swifty APIs instead of raw APIs when they exist.
```suggestion
                return v.slice(lowerBounds: begin, sizes: shape)
```